src/Tribe/Aggregator/Record/Activity.php

```diff
@@ -273,4 +275,18 @@ public function exists( $slug ) {
 		// Check if it actually exists
 		return ! empty( $this->items[ $slug ] ) ;
 	}
+
+	/**
+	 * Checks the activities for a slug to make sure there is no incoherency due to concurring processes.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $slug
+	 */
+	protected function sanity_check( $slug ) {
```

Comment:
This method name is pretty generic. Perhaps something obnoxiously descriptive like: `prevent_duplicates_between_item_actions()`

Reply:
good call, what about `kill_unicorns`? We are in fact checking for a quite remote possibility.

Reply:
Other possibilities:
* `prevent_death_and_destruction()`
* `yo_dawg_i_heard_you_like_items_so_i_put_some_items_in_your_items_but_then_i_changed_my_mind()`
* `y_u_no_work_right()`

Reply:
I like the first one. Next time.